### PR TITLE
feat: list batch job

### PR DESCRIPTION
**Pull request by srindom**

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/batch/index.ts
Comment (adrien2p) [inline packages/medusa/src/api/routes/admin/batch/index.ts:6]: As we have seen in other part of the code base, this should be in the exported function to avoid unexpected side effects with wrong usage

File: packages/medusa/src/api/routes/admin/batch/list-batch-jobs.ts
Comment (adrien2p) [inline packages/medusa/src/api/routes/admin/batch/list-batch-jobs.ts:144]: Shoudn’t  it be an emun and array?

File: packages/medusa/src/types/batch-job.ts
Comment (adrien2p) [inline packages/medusa/src/types/batch-job.ts:25]: Shouldn’t it be an array as well?

**Comment by olivermrbl** [comment]
Solves #1275

File: packages/medusa/src/api/routes/admin/batch/list-batch-jobs.ts
Comment (pKorsholm) [reply packages/medusa/src/api/routes/admin/batch/list-batch-jobs.ts:127]: the `each:true` applies the decorator to each entry in the array, but we

File: packages/medusa/src/types/batch-job.ts
Comment (pKorsholm) [reply packages/medusa/src/types/batch-job.ts:25]: same as above

File: packages/medusa/src/types/batch-job.ts
Comment (pKorsholm) [reply packages/medusa/src/types/batch-job.ts:34]: not at the moment due to how the `@IsType` is implemented. If you can get the type as `string[]` and have a great idea for parsing that we should look into it for clarity. 

The file is: `src/utils/validators/is-type`

**Review by srindom — COMMENTED** [review]
LGTM - thanks for taking over @pKorsholm! 

@olivermrbl I think we will need an approval from you as I can't approve my own PR

**Review by olivermrbl — APPROVED** [review]
LGTM!

suggestion: I'd like to see a PR description before we merge it in

**Pull request merged**
